### PR TITLE
Add importer for project KB MSR2019 dataset

### PR DESCRIPTION
**Pull request by sbs2001**
Signed-off-by: Shivam Sandbhor <shivam.sandbhor@gmail.com>

### CODE REVIEW

File: vulnerabilities/importers/project_kb_msr2019.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        return csv.reader(lines)
+
+    @staticmethod
+    def to_advisories(csv_reader):
+        # Project KB MSR csv file has no header row
+        all_advsiories = []
+        for row in csv_reader:
+            vuln_id, proj_home, fix_commit, _ = row
+            commit_link = proj_home + "/commit/" + fix_commit
+            all_advsiories.append(
```
Comment (pombredanne) [inline vulnerabilities/importers/project_kb_msr2019.py:87]: nit: all_advsiories -> all_advisories, also why not just call this "advisories" ?

File: vulnerabilities/importers/project_kb_msr2019.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        response = urllib.request.urlopen(self.url)
+        lines = [l.decode("utf-8") for l in response.readlines()]
+        return csv.reader(lines)
+
+    @staticmethod
+    def to_advisories(csv_reader):
+        # Project KB MSR csv file has no header row
+        all_advsiories = []
+        for row in csv_reader:
+            vuln_id, proj_home, fix_commit, _ = row
```
Comment (pombredanne) [inline vulnerabilities/importers/project_kb_msr2019.py:85]: What's `_` here?

File: vulnerabilities/importers/project_kb_msr2019.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            commit_link = proj_home + "/commit/" + fix_commit
+            all_advsiories.append(
+                Advisory(
+                    summary="",
+                    impacted_package_urls=[],
+                    vuln_references=[Reference(url=commit_link)],
+                    cve_id=vuln_id,
```
Comment (pombredanne) [inline vulnerabilities/importers/project_kb_msr2019.py:92]: This is likely to become "identifier", correct?

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks. See my few nits inline.
I wonder if the format will be different in "kaybee" ?

File: vulnerabilities/importers/project_kb_msr2019.py (hunk 3 above)
Comment (sbs2001) [reply vulnerabilities/importers/project_kb_msr2019.py:92]: yes

File: vulnerabilities/importers/project_kb_msr2019.py (hunk 2 above)
Comment (sbs2001) [reply vulnerabilities/importers/project_kb_msr2019.py:85]: https://github.com/SAP/project-kb/blob/master/MSR2019/dataset/vulas_db_msr2019_release.csv

It's `pos` for all rows. No idea what it means.

File: vulnerabilities/importers/project_kb_msr2019.py (hunk 1 above)
Comment (sbs2001) [reply vulnerabilities/importers/project_kb_msr2019.py]: Done

File: vulnerabilities/importers/project_kb_msr2019.py (hunk 2 above)
Comment (pombredanne) [reply vulnerabilities/importers/project_kb_msr2019.py:85]: Note that FWIW, you can always write:
`for vuln_id, proj_home, fix_commit, _ in csv_reader:`

File: vulnerabilities/importers/project_kb_msr2019.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        lines = [l.decode("utf-8") for l in response.readlines()]
+        return csv.reader(lines)
+
+    @staticmethod
+    def to_advisories(csv_reader):
+        # Project KB MSR csv file has no header row
+        advsiories = []
+        for row in csv_reader:
+            vuln_id, proj_home, fix_commit, _ = row
+            commit_link = proj_home + "/commit/" + fix_commit
```
Comment (pombredanne) [inline vulnerabilities/importers/project_kb_msr2019.py:86]: Nit: I would tend to prefer a .format or an f-string.

**Review by pombredanne — APPROVED** [review]
All good. See some FYI.

**Pull request merged**